src/apps/file/api/file.py

```diff
@@ -0,0 +1,52 @@
+import uuid
+
+from botocore.exceptions import ClientError  # type: ignore
+from fastapi import Body, Depends, File, UploadFile
+from fastapi.responses import StreamingResponse
+
+from apps.authentication.deps import get_current_user
+from apps.file.domain import FileDownloadRequest, UploadedFile
+from apps.file.errors import FileNotFoundError
+from apps.shared.domain.response import Response
+from apps.users.domain import User
+from config import settings
+from infrastructure.utility.cdn_client import CDNClient
+
+
+async def upload(
+    file: UploadFile = File(...),
+    user: User = Depends(get_current_user),
+) -> Response[UploadedFile]:
+    cdn_client = CDNClient(settings.cdn, env=settings.env)
+
+    # generate key
+    key = CDNClient.generate_key(str(uuid.uuid4()), file.filename)
+    print(file.content_type)
+
+    # upload file
+    try:
+        cdn_client.upload(key, file.file)
+    except Exception as e:
+        raise e
```

Comment:
This construction makes no sense.
All exceptions are handled on middleware level

```suggestion
    # upload file

    cdn_client.upload(key, file.file)
```